### ISSUE TITLE
Fully Portable Badge Assertions with Embedded BadgeClass and Issuer

### ISSUE THREAD

**Issue by timothyfcook** [openbadges/openbadges-specification#82]
Use the Linked Data conventions to enable badge resources to contain full representations of their key related objects as well allowing for the use of lightweight IRI-based pointers.

**User Stories**: 
1. As a badge issuer, I want to embed the full metadata of the BadgeClass and Issuer Profile into an Assertion, so that I can cryptographically sign the entirety of the metadata associated with it and bake the entire object into a portable image format.
2. As a badge recipient, I want my badge to contain all relevant information related to that badge so that it does rely on linked references that can become deprecated.
3. As a badge consumer, I want to understand when the BadgeClass or Issuer Profile included in two different Assertions are referencing the same entity.

**Comment by ottonomy** [comment]
Note: For future-compatibility, the IRIs used as `@id`s of BadgeClasses and Issuer profiles may not have to be HTTP-based/resolvable by browsers. They may just be a unique identifier if the public key of the issuer is trusted.

**Comment by ottonomy** [comment]
Prototype:

Current 1.x badges use a "badge" property to identify a URL (IRI) of an associated BadgeClass.

```jsonld
{
    "@context": "https://w3id.org/openbadges/v1",
    "badge": "https://example.com/badgeclass"
}
```

Linked Data allows substitution of a document for such a IRI reference. In the badges context "id" is an alias for the JSON-LD keyword "@id", so the document below is a fully portable declaration that still carries the canonical identifier. 

```jsonld
{
    "@context": "https://w3id.org/openbadges/v1",
    "badge": {
        "id": "https://example.com/badgeclass",
        "name": "Expert of Robotics",
        "description": "Earners of this badge have become robotics experts",
        "issuer": {
            "id": "https://example.com/issuer",
            "name": "Example Issuer",
            "url": "https://example.com"
        }
    }
}
```

**Comment by mgylling** [comment]
Closing this issue, as it was resolved by the 2.0 Recommendation. This issue and all other issues resolved by 2.0 can be found via the 2.0 Release Candidate milestone and PR #99.
